cpp/src/lists/interleave_columns.cu

```diff
@@ -169,6 +169,10 @@ struct compute_string_sizes_and_interleave_lists_fn {
     auto const start_str_idx = list_offsets[list_id];
     auto const end_str_idx   = list_offsets[list_id + 1];
 
+    // In case of empty list (i.e. it doesn't contain any string element), we just ignore it because
+    // there will not be anything to store for that list in the child column.
+    if (start_str_idx == end_str_idx) { return; }
```

Comment:
Special cases like this are always a code smell to me that a more appropriate algorithm isn't being used.  

Ideally, any algorithm should implicitly handle an empty input.

Looking below, the first code path handles this:
```
for (auto read_idx = start_str_idx; read_idx < end_str_idx; ++read_idx, ++write_idx) {
```
This won't execute any iterations.

```
      auto const start_byte = str_offsets[start_str_idx];
      auto const end_byte   = str_offsets[end_str_idx];
```

Is this the problem? `str_offsets` may be nullptr?

Reply:
Yes, this is. In case of empty (all empty) lists, `str_offsets` from the child column will be empty too.

Reply:
In fact, I remember to add tests with empty lists at the beginning. However, I didn't consider the case where a lists column input is not empty but contains all empty lists.